homeassistant/components/google/calendar.py

```diff
@@ -213,24 +224,76 @@ def append_calendars_to_config() -> None:
         )
 
 
-class GoogleCalendarEntity(CalendarEntity):
-    """A calendar event device."""
+class CalendarUpdateCoordinator(DataUpdateCoordinator):
+    """Coordinator for calendar RPC calls."""
 
     _attr_has_entity_name = True
 
     def __init__(
         self,
+        hass: HomeAssistant,
         calendar_service: GoogleCalendarService,
+        name: str,
+        calendar_id: str,
+        search: str | None,
+    ) -> None:
+        """Create the Calendar event device."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=name,
+            update_interval=MIN_TIME_BETWEEN_UPDATES,
+        )
+        self.calendar_service = calendar_service
+        self.calendar_id = calendar_id
+        self._search = search
+
+    async def async_get_events(
+        self, start_date: datetime, end_date: datetime
+    ) -> list[Event]:
+        """Get all events in a specific time frame."""
+        request = ListEventsRequest(
+            calendar_id=self.calendar_id,
+            start_time=start_date,
+            end_time=end_date,
+            search=self._search,
+        )
+        result_items = []
+        try:
+            result = await self.calendar_service.async_list_events(request)
+            async for result_page in result:
+                result_items.extend(result_page.items)
+        except ApiException as err:
+            _LOGGER.error("Unable to connect to Google: %s", err)
+            return []
```

Comment:
I think it would be better to allow calendar platforms to raise HomeAssistantError if getting events fails from the api. Ie catch HomeAssistantError in the endpoint view handler.

Reply:
Yes, agreed -- also silently failing seems incorrect.  Let me address this first in a separate PR since it's on my list to fix anyway.

Reply:
I've sent https://github.com/home-assistant/core/pull/74891 to address this, then i'll revisit this PR.

Reply:
I've rebased and merged to address this.